java/ql/src/semmle/code/java/dataflow/FlowSources.qll

```diff
@@ -318,3 +319,34 @@ class AndroidIntentInput extends DataFlow::Node {
     )
   }
 }
+
+/** Method access to external inputs of `android.content.Intent` or `android.os.BaseBundle` object. */
+class IntentGetExtraMethodAccess extends MethodAccess {
+  IntentGetExtraMethodAccess() {
+    exists(AndroidComponent ac |
+      this.getEnclosingCallable().getDeclaringType() = ac and
+      ac.isExported() and
+      this.getMethod().getName().regexpMatch("get\\w+Extra") and
+      this.getMethod().getDeclaringType() instanceof TypeIntent
+    )
+    or
+    this.getMethod().getName().regexpMatch("get\\w+") and
```

Comment:
Split these two (get***Extra and the Bundle getters), since they're pretty unrelated

Reply:
With the <code>AndroidIntentInput</code> change, only Bundle getters modelled in <code>GetBundleExtraMethodAccess</code> are required as the additional taint step.